gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/iceberg/IcebergTableMetadataValidatorTest.java

```diff
@@ -0,0 +1,165 @@
+  private static final Schema schema2 = AvroSchemaUtil.toIceberg(avroDataSchema2);
+  private static final Schema schema3 = AvroSchemaUtil.toIceberg(avroDataSchema3);
+  private static final Schema schema4 = AvroSchemaUtil.toIceberg(avroDataSchema4);
+  private static final PartitionSpec partitionSpec1 = PartitionSpec.builderFor(schema1)
+      .identity("field1")
+      .build();
+  private static final String SHOULD_NOT_THROW_EXCEPTION = "Should not throw any exception";
+  private static final String SCHEMA_MISMATCH_EXCEPTION = "Schema Mismatch between Source and Destination Iceberg Tables Schema";
+  private static final String PARTITION_SPEC_MISMATCH_EXCEPTION = "Partition Spec Mismatch between Source and Destination Iceberg Tables Partition Spec";
+  private static final String PARTITION_SPEC_DIFF_TYPE_EXCEPTION = "Partition Spec Have different types for same partition field between Source and Destination Iceberg Table";
+
+  @Test
+  public void testValidateSameSchema() {
+    try {
+      IcebergTableMetadataValidator.validateSchemaForEquality(schema1, schema1);
+    } catch (Exception e) {
+      System.out.println(e.getMessage());
+      Assert.fail(SHOULD_NOT_THROW_EXCEPTION);
+    }
+  }
+
+  @Test
+  public void testValidateDifferentSchema() {
+    // Schema 1 and Schema 2 have different field order
+    verifyValidateSchemaIllegalArgumentException(schema1, schema2);
+  }
+
+  @Test
+  public void testValidateSchemaWithDifferentTypes() {
+    verifyValidateSchemaIllegalArgumentException(schema3, schema4);
+  }
+
+  @Test
+  public void testValidateSchemaWithEvolvedDestinationSchema() {
+    // TODO: This test should pass in the future when we support source side schema evolution and here there should
+    //  not be any commit needed on destination side
+    // Schema 3 has one more extra field as compared to Schema 1
+    verifyValidateSchemaIllegalArgumentException(schema1, schema3);
+  }
+
+  @Test
+  public void testValidateSchemaWithEvolvedSourceSchema() {
+    // TODO: This test should pass in the future when we support source side schema evolution and commit the changes
+    //  on destination side either through IcebergRegisterStep or any other CommitStep
+    // Schema 3 has one more extra field as compared to Schema 1
+    verifyValidateSchemaIllegalArgumentException(schema3, schema1);
+  }
+
+  @Test
+  public void testValidateSamePartitionSpec() {
+    try {
+      IcebergTableMetadataValidator.validatePartitionSpecForEquality(partitionSpec1, partitionSpec1);
+    } catch (Exception e) {
+      System.out.println(e.getMessage());
+      Assert.fail(SHOULD_NOT_THROW_EXCEPTION);
+    }
+  }
+
+  @Test
+  public void testValidatePartitionSpecWithDiffType() {
+    PartitionSpec partitionSpec4 = PartitionSpec.builderFor(schema4)
+        .identity("field1")
+        .build();
+    verifyValidatePartitionSpecIllegalArgumentException(partitionSpec1, partitionSpec4, PARTITION_SPEC_DIFF_TYPE_EXCEPTION);
```

Comment:
let's use `assertThrows`

Reply:
`verifyValidatePartitionSpecIllegalArgumentException` function does expect throws and validate the thrown message as well.
Have added it to verify both error messages are thrown based on condition check.